pkg/encoding/v1/encoding.go

```diff
@@ -184,6 +184,7 @@ func (c *Container) UnmarshalYAML(unmarshal func(interface{}) error) error {
 type Service struct {
 	Name       ResourceName `yaml:"name"`
 	Containers []Container  `yaml:"containers"`
+	Replicas   int32        `yaml:"replicas,omitempty"`
```

Comment:
it has to be a pointer here otherwise you can't distinguish between user setting 0 and not specifying the value